src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -842,6 +842,16 @@ public function providePathsForMakePathRelative()
             array('/a/aab/bb/', '/b/aab', '../../a/aab/bb/'),
             array('/aab/bb', '/aa', '../aab/bb/'),
             array('/aab', '/aa', '../aab/'),
+            array('/aa/bb/cc', '/aa/dd/..', 'bb/cc/'),
+            array('/aa/../bb/cc', '/aa/dd/..', '../bb/cc/'),
+            array('/aa/bb/../../cc', '/aa/../dd/..', 'cc/'),
+            array('../aa/bb/cc', '/aa/dd/..', 'bb/cc/'),
+            array('../../aa/../bb/cc', '/aa/dd/..', '../bb/cc/'),
```

Comment:
Agree. Also this doesn't make sense. If the target path is already relative, what is it relative against? So just removing the leading dot segments changes the path as it's not clear what to refers to. So either you throw an exception that this is not accepted, or you can just return the relative target path as-is, since it is already relative. The latter is what I did in https://github.com/guzzle/psr7/blob/master/src/UriResolver.php#L149